### ISSUE TITLE
[FEATURE] Always up to date version of a runner

### ISSUE THREAD

**Issue by rokam** [Vysp3r/ProtonPlus#135]
**Describe the solution you'd like**
Be able to have a runner's path that contains the latest version there. Something like Proton Experimental on Steam.
That way, the game that uses that runner will always use the up-to-date version.

Also, this will require a background process to keep it up to date.

**Explanation**
Download ProtonGE Updated, this will add the most recent version to `~/.steam/steam/compatibilitytools.d/ProtonGE\ Updated/`.

And `ProtonGE Updated` should appear in the combobox:
![image](https://github.com/Vysp3r/ProtonPlus/assets/2856518/6862f366-f22c-4328-92c5-b1e5d1b7ad95)

**Comment by Vysp3r** [comment]
Yeah that should be do-able. I'll need to learn how to make a background process and how to start an application when the computer opens for that. 

One problem that I may face is let's say there's a new version available, but you're currently playing a game on Steam. How am I supposed to know you're playing? I could ask you for your Steam username and use the steam API to check if you're playing, but what if you put yourself in offline mode. Do you have any idea on how to fix that problem?

**Comment by rokam** [comment]
What about just check process running?
When running proton-ge I do get this:
```
$ ps -aux
python3 /usr/share/steam/compatibilitytools.d/proton-ge-custom/proton waitforexitandrun /jogos/SteamLibrary/steamapps/common/Fall Guys/FallGuys_client.exe
```

**Comment by Vysp3r** [comment]
Oh yeah you're right. There's nothing stopping me now. It might take some time since I need to learn new things.

**Comment by RouHim** [comment]
An alternative implementation could be that it can be updated via the command line. Something like "protonplus update all" or "protonplus update proton wine"
This would also allow integration into programs like [topgrade](https://github.com/topgrade-rs/topgrade).

**Comment by Vysp3r** [comment]
I still have some fixing to do for the next release (v0.4.7), but once I am done with it I was planning on working on this request along side https://github.com/Vysp3r/ProtonPlus/issues/148. After that I don't mind adding command lines for different stuff.

**Comment by Vysp3r** [comment]
I know it's been a long time and I am sorry for that.
I will try to give this a try after I am done with my current list of things to do.

**Comment by Vysp3r** [comment]
Sorry for the long wait, but now is finally the time.
This is the next feature on my list.
Starting to work on it right now.

**Comment by Vysp3r** [comment]
I am nearly there!

<img width="1072" height="781" alt="Image" src="https://github.com/user-attachments/assets/0ce7502f-729c-4ddd-ab4e-9ec19830e344" />

**Comment by Vysp3r** [comment]
I've created a separate issue for the command line support just not to make the scope of this one too big.

https://github.com/Vysp3r/ProtonPlus/issues/463

**Comment by Vysp3r** [comment]
I will need to implement some temporary download directory and then move the runners over because if I try to download the latest Proton-GE and you have it installed manually it deletes it which is not the behavior I want.
This will take a bit more time, but it's slowly getting there.

**Comment by Vysp3r** [comment]
As of now everything seems good.

It shows a toast notification saying it's checking for updates when you launch the application.

Depending on the result of the check, it displays another toast notification saying if it was successful or not.

If you already had a version installed and the update fails it puts back the previous version as a fallback.

Runners are now also downloaded in ```~/.cache/ProtonPlus``` or ```/.var/app/com.vysp3r.ProtonPlus/cache/ProtonPlus/``` if running as a Flatpak and they're then moved the to proper directory.

This changes also makes it so if anything bad happens and there's leftover files you can easily clean them and it's they're not left polluting the launchers runner directories.

Now it's mostly testing.
